### PR TITLE
interim ESP32-S3 changes & diagnostics

### PR DESCRIPTION
**Pull request by gojimmypi**
# Description

This update adds some minor, interim changes to the Espressif-specific library to support the ESP32-S3 as related to https://github.com/wolfSSL/wolfssl/issues/5948. 

The [Espressif wolfssl_test example ](https://github.com/wolfSSL/wolfssl/tree/master/IDE/Espressif/ESP-IDF/examples/wolfssl_test) also has new [optional git version information](https://github.com/gojimmypi/wolfssl/blob/b04d6ed56bfb260ce94dff9ebccdbbed0b5baa46/IDE/Espressif/ESP-IDF/examples/wolfssl_test/main/main.c#L141) as noted in https://github.com/wolfSSL/wolfssl/pull/5955

See also the new `VisualGDB_wolfssl_test_IDF_v5_ESP32S3.vgdbproj` VisualGDB file.

Add this to the end of `user_settings.h` to detect and disable hardware acceleration for the ESP32-S3:

```
#if defined(CONFIG_IDF_TARGET_ESP32C3)
    #define NO_ESP32WROOM32_CRYPT
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_HASH
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_AES
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI
#elif  defined(CONFIG_IDF_TARGET_ESP32S3)
    #define NO_ESP32WROOM32_CRYPT
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_HASH
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_AES
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI
#elif  defined(CONFIG_IDF_TARGET_ESP_OTHER)

#else

#endif
```

Fixes zd# n/a

# Testing

Run the `./setup.sh` for Espressif component install as explained in the [Setup for Linux](https://github.com/wolfSSL/wolfssl/blob/master/IDE/Espressif/ESP-IDF/README.md#setup-for-linux).

This is not full ESP32-S3 support. Other ESP32-S3 changes are currently WIP on the [Espressif_Multichip](https://github.com/gojimmypi/wolfssl/tree/Espressif_Multichip)

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: IDE/Espressif/ESP-IDF/examples/wolfssl_test/main/main.c [hunk 1]
```diff
@@ -126,6 +128,44 @@ void my_atmel_free(int slotId)
...
+
+#if defined(LIBWOLFSSL_VERSION_GIT_HASH_DATE)
+    ESP_LOGI(TAG, "LIBWOLFSSL_VERSION_GIT_HASH_DATE = %s", LIBWOLFSSL_VERSION_GIT_HASH_DATE);
+#endif
+
+
+#if defined(CONFIG_IDF_TARGET_ESP32C3)
+    /* not available for C3 at this time */
+#else
+   // ESP_LOGI(TAG, "CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ = %u MHz", CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ);
```
Comment (dgarske) [inline IDE/Espressif/ESP-IDF/examples/wolfssl_test/main/main.c:157]: Please cleanup this line.

**Review by dgarske — CHANGES_REQUESTED** [review]
Holding merge till #5955 is merged.

File: IDE/Espressif/ESP-IDF/examples/wolfssl_test/main/main.c (hunk 1 above)
Comment (gojimmypi) [reply IDE/Espressif/ESP-IDF/examples/wolfssl_test/main/main.c]: As yes, good. Thanks for finding that. That was not intended.

I've added something a bit more clear and graceful, although not yet the final product to fully and properly support all of the chipsets. This should bring operational support without HW acceleration for the ESP32-S3 (cc: @PaulMartinsen)

btw - note there's a bit of chicken-and-egg issue with the wolfSSL `user_settings.h`  and target Espressif chip selection at this time. My user settings has this logic at the tail of of my file to detect the Espressif target from the `sdkconfig.h` file and _conditionally_ turn off HW acceleration:

```
#include "sdkconfig.h"

/* adjust wait-timeout count if you see timeout in rsa hw acceleration */
#define ESP_RSA_TIMEOUT_CNT    0x249F00
#if defined(CONFIG_IDF_TARGET_ESP32C3)
    #define NO_ESP32WROOM32_CRYPT
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_HASH
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_AES
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI
#elif defined(CONFIG_IDF_TARGET_ESP32S3)
    #define NO_ESP32WROOM32_CRYPT
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_HASH
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_AES
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI
#elif defined(CONFIG_IDF_TARGET_ESP_OTHER)
   /* other */
#else
  /* default is ESP32-WROOM */
#endif
```

One can of course simply ensure hardware acceleration is _always_ off in the `user_settings.h` without detecting architecture. 

Without turning off hardware acceleration for things like the `ESP32-S3` at this time, there will be unexpected and unintuitive compile-time errors.

Also note during this update I saw what appeared to be runtime memory corruption not detected at flash time when doing this:

```
idf.py  -b 921600  -p /dev/ttyS8  build flash monitor
```

I fixed it by doing a `idf.py fullclean`, deleting the existing `sdkconfig` and then loading at a slower [baud rate](https://docs.espressif.com/projects/esptool/en/latest/esp32/esptool/basic-options.html#baud-rate):

```
idf.py  -b 230400  -p /dev/ttyS8 set-target esp32-s3 build flash monitor
```

And for the regular ESP32-WROOM32:

```
idf.py  -b 230400  -p /dev/ttyS15 set-target esp32 build flash monitor
```

Also - I added the known-good `sdkconfig.defaults` currently used in my [WIP Espressif multichip branch](https://github.com/gojimmypi/wolfssl/tree/Espressif_Multichip/IDE/Espressif/ESP-IDF/examples/wolfssl_test).

The ESP32 `wolfssl_test` was confirmed to work with both the ESP32 and ESP32-S3.

**Comment by gojimmypi** [comment]
Curiously the last wolfSSL build check for "Test Espressif  examples" validation created by @bandi13 failed even though there was no code change. It was only by a small app size amount, but nonetheless a failure.

In response, I've changed all 4 of the local examples to now have "large app" settings in `sdkconfig.default`. See [24f8933](https://github.com/wolfSSL/wolfssl/pull/5956/commits/24f89337f6b537cfba0a354ae86fac2fbb3c9da2).

**Review by dgarske — APPROVED** [review]

**Pull request merged**